### DISCUSSION TITLE
Should I pip install poetry ?

### DISCUSSION THREAD

**Discussion by aahnik** [python-poetry/poetry#3946] [Q&A] [answered]
The official documentation warns not to.

But should I?

I see that these projects are pip installing poetry:

https://github.com/PyCQA/isort/blob/f074239e5e39c2dd9e4d373e3c6ce528dae3f78c/Dockerfile#L5

https://github.com/abatilo/actions-poetry/blob/8284d202bc272a8d0597e26e1c0b4a0d0c73db93/action.yml#L16

> Be aware that it will also install Poetry's dependencies which might cause conflicts with other packages.

how can that happen ? my application/library, and its dependencies will be installed in a virtual env, but poetry is installed at user level (outside of any virtual env)

if, in my user level, I have no other python packages to worry about (like in a Docker container, or a github action), then can I pip install poetry ?

**Comment by abn** [comment, marked as answer]
It really depends on your use case. 

If this is for your workstation, I would generally advice using either the `install-poetry.py` script, or [`pipx`](https://pipxproject.github.io/pipx/) (personal favorite). The rationale for this is that when using your system-site (or user-site), you are installing dependencies along with the package into a shared site, this means installing package A (1.0.2) might install package B (1.0.0); and then later installing package C (3.0.0) might install package B (0.5.0) because it is not compatible with B (1.0.0). This might cause package A to fail. These sort of conflicts are more common in longer lived and/or shard purpose environments; like developer machines. This ensures poetry does nto break when you upgrade, remove or install other things via `pip` or your system's package manager.

If you are installing poetry on an ephemeral CI environment or in a container, installing to the system-site might cause minimal side-effects and should be okay. But my advice in these scenarios (or well any scenario), is to ensure that you **do not** disable virtual environment creation in poetry (even in a container). This would mean that when you run a poetry command for a project, a new virtual environment will be created if one does not already exist, ensuring that the project's dependencies are isolated and clean. If you choose not to use virtual environments, you can end up in a scenario where poetry will decide to remove unused dependencies or upgrade dependencies leaving you with a broken system-site when managing dependencies for a project.

But all in all, poetry is a tool. It is upto you to use it in a way that makes sense for you.

**Reply by aahnik** [reply]
❤️ poetry's support via gh discussions is awesome 🙏
